src/hugr/validate.rs

```diff
@@ -27,6 +28,8 @@ struct ValidationContext<'a> {
     hugr: &'a Hugr,
     /// Dominator tree for each CFG region, using the container node as index.
     dominators: HashMap<Node, DominatorTree>,
+    /// Resource requirements associated with each edge
+    resources: HashMap<(Node, Port), ResourceSet>,
```

Comment:
I didn't see it before, but will same-direction ports in the same node ever have different ResourceSets? Would `(Node, Direction)` work?

Reply:
Hmm I guess it would, I was thinking using the specific port offers the flexibility of knowing where to put a lift node in some cases

Reply:
If it's useful leave it as is. It's a trade-of between memory/perf and flexibility.

Reply:
I think I'll change it to `(Node, Direction)` after all

Reply:
`(Node, Direction)` will require more Lift nodes I think, as with `(Node, Port)` there is an implicit union over inputs? I guess the ideal would be `(Node, Port)` for inputs and `Node` for outputs....

Reply:
> `(Node, Direction)` will require more Lift nodes I think, as with `(Node, Port)` there is an implicit union over inputs? I guess the ideal would be `(Node, Port)` for inputs and `Node` for outputs....

I don't think this applies here, as we're not trying to add lift nodes